File: ecom-project-backend/user-service/src/main/java/org/ecommerce/user/entities/Address.java
package org.ecommerce.user.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;


@Entity
@Data
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Size(min = 2, message = "Address name should have least 2 characters")
    private String addressName;
    private String apartment;

    private String area;

    private String landmark;

    @Size(min = 6, message = "Pin code should be of 6 digits")
    @NotBlank(message = "Pincode cannot be empty")
    private String pincode;

    private String city;
    private String state;

    @ManyToOne
    @JsonIgnore
    private User user;


    @Override
    public String toString() {
        return "Address - " +
                addressName + ", " +
                apartment + ", " +
                area + ", " +
                landmark + ", " +
                pincode + ", " +
                city + ", " +
                state
                ;
    }
}

File: ecom-project-frontend/cart-mf/src/components/Address/Address.js
import "./Address.css";

const Address = ({ address, selected }) => {
  return (
    <div className={`address card my-2 ${selected ? "selected" : ""}`}>
      <div className="card-body px-3 py-2">
        <p className="address-title m-0 py-2 ">{address?.addressName}</p>
        <div className="address-text d-flex flex-wrap gap-2">
          <p>{address?.apartment},</p>
          <p>{address?.area ? address?.area + "," : ""}</p>
          <p>{address?.landmark ? address?.landmark + "," : ""}</p>
          <p>{address?.city ? address?.city + "," : ""}</p>
          <p>{address?.state ? address?.state : ""} - </p>
          <p>{address?.pincode}</p>
        </div>
      </div>
    </div>
  );
};

export default Address;
